### PR TITLE
pythonPackages.annexremote: init at 1.3.1

### PR DESCRIPTION
**Pull request by montag451**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/annexremote/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "058d39fd59db165f0014601da1ba9ecde7614de87eef3061e9ea0b1dd096a9cd";
+  };
+
+  propagatedBuildInputs = [ future ];
+
+  meta = with stdenv.lib; {
+    description = "git annex special remotes made easy";
+    homepage = https://pypi.org/project/annexremote/;
```
Comment (dotlambda) [inline pkgs/development/python-modules/annexremote/default.nix]:
```suggestion
    homepage = https://github.com/Lykos153/AnnexRemote;
```

File: pkgs/development/python-modules/annexremote/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  version = "1.3.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "058d39fd59db165f0014601da1ba9ecde7614de87eef3061e9ea0b1dd096a9cd";
+  };
+
+  propagatedBuildInputs = [ future ];
+
+  meta = with stdenv.lib; {
```
Comment (dotlambda) [inline pkgs/development/python-modules/annexremote/default.nix]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/python-modules/annexremote/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ stdenv
```
Comment (dotlambda) [inline pkgs/development/python-modules/annexremote/default.nix]:
```suggestion
{ lib
```

File: pkgs/development/python-modules/annexremote/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "058d39fd59db165f0014601da1ba9ecde7614de87eef3061e9ea0b1dd096a9cd";
+  };
+
+  propagatedBuildInputs = [ future ];
+
+  meta = with stdenv.lib; {
+    description = "git annex special remotes made easy";
```
Comment (dotlambda) [inline pkgs/development/python-modules/annexremote/default.nix]:
```suggestion
    description = "Helper module to easily develop git-annex remotes";
```
seems like a better description and is capitalized as required.

**Review by dotlambda — CHANGES_REQUESTED** [review]
> Ran 0 tests in 0.000s

Please add an appropriate `checkPhase`. You might have to use `fetchFromGitHub`. In that case, please ask upstream to include the tests in their PyPI tarballs in the future.

**Review by dotlambda — CHANGES_REQUESTED** [review]
The commit message needs to be corrected.

**Comment by montag451** [comment]
@dotlambda I tried to address all your remarks in 5cc82aaa6384092bce08bdbb6a3ffa6541aa46b6

File: pkgs/development/python-modules/annexremote/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+, fetchFromGitHub
+, future
+, nose
+}:
+
+buildPythonPackage rec {
+  pname = "annexremote";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
```
Comment (dotlambda) [inline pkgs/development/python-modules/annexremote/default.nix]: Please add a comment with the reason for not using `fetchPypi`.

Comment (montag451) [reply pkgs/development/python-modules/annexremote/default.nix]: See 48a5d481baccb2d49305ed863b45c57f92ac6cd0

**Comment by dotlambda** [comment]
> In that case, please ask upstream to include the tests in their PyPI tarballs in the future.

I did this for you in https://github.com/Lykos153/AnnexRemote/pulls.

> The commit message needs to be corrected.

I'll also do that for you when squashing.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.annexremote python3.pkgs.annexremote

**Review by dotlambda — CHANGES_REQUESTED** [review]
> error: build of '/nix/store/l3dpg0z5qmnddgz67vrhjg9anmxq3cnv-python2.7-annexremote-1.3.1.drv' failed

**Comment by montag451** [comment]
Build failure fixed. See 7a7ec744d7fbd7c29b61978c0ad760b7935d6505

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.annexremote python3.pkgs.annexremote

**Pull request merged**